src/librustc/middle/traits/error_reporting.rs

```diff
@@ -290,7 +290,7 @@ pub fn maybe_report_ambiguity<'a, 'tcx>(infcx: &InferCtxt<'a, 'tcx>,
                     {
                         span_err!(infcx.tcx.sess, obligation.cause.span, E0282,
                                 "unable to infer enough type information about `{}`; \
-                                 type annotations required",
+                                 type annotations or generic parameter binding required",
```

Comment:
@pnkfelix It were definitely those tests that failed. To be sure, I ran `make check` on `414dfb1` on my Windows box again. There were indeed failures, I missed them the first time. I did see ` ... FAILED` scroll by this time, but the check does not stop on failure. The last few lines of output are these:

```
test [run-pass] run-pass/xcrate-unit-struct.rs ... ok
test [run-pass] run-pass/zero-size-type-destructors.rs ... ok
test [run-pass] run-pass/zero_sized_subslice_match.rs ... ok

test result: ok. 2034 passed; 0 failed; 33 ignored; 0 measured

make: *** wait: No child processes.  Stop.
```
Last time I only checked the last line, and I assumed `0 failed` meant everything was good. The summary of compile-fail turned out to be hidden 2000 lines earlier between the rest of the test output.